apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx

```diff
@@ -72,77 +70,111 @@ const VersionHistoryDropdown: React.FunctionComponent<
     closeDropdown();
   }, [dispatch, startSource, updatedSourceCallback, closeDropdown]);
 
+  const restoreSelectedVersion = useCallback(() => {
+    const projectManager = Lab2Registry.getInstance().getProjectManager();
+    if (selectedVersion === INITIAL_VERSION_ID) {
+      startOver();
+    } else if (projectManager && selectedVersion) {
+      setLoading(true);
+      setLoadError(false);
+      projectManager
+        .restoreSources(selectedVersion)
+        .then(sources => {
+          if (sources) {
+            dispatch(setProjectSource(sources));
+            if (updatedSourceCallback) {
+              updatedSourceCallback(sources);
+            }
+          } else {
+            setLoadError(true);
+          }
+          setLoading(false);
+          closeDropdown();
+        })
+        .catch(() => {
+          setLoadError(true);
+          setLoading(false);
+        });
```

Comment:
pretty minor, but I'd bump the `setLoading(false)` into a `finally` block so it's not duplicated across cases.

Reply:
done!